src/FilesystemRepository.php

```diff
@@ -124,10 +119,10 @@ public static function isSymlinkSupported()
      */
     public function __construct($baseDir = '/', $symlink = true, $relative = true)
     {
-        Assert::directory($baseDir);
+        parent::__construct($baseDir);
+
         Assert::boolean($symlink);
 
-        $this->baseDir = rtrim(Path::canonicalize($baseDir), '/');
```

Comment:
Hm, this and the PathMappingRepository do two different things now. PathMappingRepository restricts newly added resources to be within the given base directory. The FilesystemRepository, on the other hand, moves added files to that directory - but the added files can be located anywhere.

Does it make sense to also restrict resources added to the FilesystemRepository to be within a specific base directory?